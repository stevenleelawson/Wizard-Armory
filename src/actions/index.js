export const loadCards = (cards, color) => ({
  type: 'LOAD_CARDS',
  cards,
  color
});

export const formState = (color) => ({
  type: 'FORM_STATE',
  color
});

export const addToDeck = (card) => ({
  type: 'ADD_TO_DECK',
  card
});

export const clearDeck = () => ({
  type: 'CLEAR_DECK'
})
