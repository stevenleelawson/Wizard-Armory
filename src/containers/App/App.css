@import url('https://fonts.googleapis.com/css?family=Lora');

.App {
  text-align: center;
}


.App-header {
  background-image: url('./battlinwizard.jpg');
  background-size: cover;
  height: 150px;
  padding: 20px;
  color: #4C1502;
  font-family: 'Lora', serif;
  display: flex;
  flex-direction: column;
  border: .5px solid black;
}

.App-title {
  font-size: 3em;
  margin-top: 0
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
