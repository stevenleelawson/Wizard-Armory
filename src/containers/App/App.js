import React, { Component } from 'react';
import * as api from '../../apiCalls';
import * as actions from '../../actions';
import { connect } from 'react-redux';
import { Route, withRouter } from 'react-router-dom';
import './App.css';
import CardContainer from '../CardContainer';
import DeckContainer from '../DeckContainer';
import Form from '../Form';
import PropTypes from 'prop-types';
import SignUpForm from '../../components/SignUpForm';
import SignInForm from '../../components/SignInForm';
import Navigation from '../../components/Navigation';
import { firebase } from '../../firebase';

export class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      authUser: null
    };
  }

  componentDidMount() {
    firebase.auth.onAuthStateChanged(authUser => {
      authUser
        ? this.setState(() => ({ authUser }))
        : this.setState(() => ({ authUser: null}));
    });

  }

  componentDidUpdate(prevProps, nextProps) {
    if (prevProps !== nextProps) {
      this.fetchCards();
    }
  }

  fetchCards = async () => {
    const color = this.props.color;

    if (!this.props.cards[color]) {
      const cards = await api.getCards(this.props.color);
      this.props.loadCards(cards, color);
    }
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">WIZARD ARMORY</h1>
          <Route exact path='/' component={Form} />
          <Navigation authUser={this.state.authUser} />
        </header>
        <Route exact path='/' component={CardContainer} />
        <Route exact path='/deck' component={DeckContainer} />
        <Route exact path='/signup' component={SignUpForm} />
        <Route exact path='/signin' component={SignInForm} />
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  color: state.color,
  cards: state.cards
});

export const mapDispatchToProps = dispatch => ({
  loadCards: (cards, color) => dispatch(actions.loadCards(cards, color)),
  formState: color => dispatch(actions.formState(color))
});


export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App));

App.propTypes = {
  color: PropTypes.string,
  loadCards: PropTypes.func,
  cards: PropTypes.oneOfType([
    PropTypes.object,
    PropTypes.array
  ])
};
