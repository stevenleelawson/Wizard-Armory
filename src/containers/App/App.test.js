import React from 'react';
import { App } from './App';
import { shallow } from 'enzyme';
import * as actions from '../../actions';


jest.mock('../../apiCalls');

describe('App', () => {
  let mockLoadCards;
  beforeEach( () => {
    mockLoadCards = jest.fn();
  });

  it('should match the snapshot', () => {
    const wrapper = shallow(
      <App
        loadCards={mockLoadCards}
      />,
      { disableLifecycleMethods: true }
    )
    expect(wrapper).toMatchSnapshot();
  });
});
