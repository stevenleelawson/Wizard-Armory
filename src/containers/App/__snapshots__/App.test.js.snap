// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    loadCards={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="App-header"
>
          <h1
                    className="App-title"
          >
                    Wizard Tracker
          </h1>
</header>,
        <Route
          path="/"
          render={[Function]}
/>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1
            className="App-title"
>
            Wizard Tracker
</h1>,
          "className": "App-header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Wizard Tracker",
            "className": "App-title",
          },
          "ref": null,
          "rendered": "Wizard Tracker",
          "type": "h1",
        },
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "path": "/",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="App-header"
>
            <h1
                        className="App-title"
            >
                        Wizard Tracker
            </h1>
</header>,
          <Route
            path="/"
            render={[Function]}
/>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1
              className="App-title"
>
              Wizard Tracker
</h1>,
            "className": "App-header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Wizard Tracker",
              "className": "App-title",
            },
            "ref": null,
            "rendered": "Wizard Tracker",
            "type": "h1",
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
